client/app/bundles/course/assessment/submission/components/GetHelpChatPage/ChipButton.tsx

```diff
@@ -0,0 +1,121 @@
+  fetchLiveFeedbackStatus,
+} from '../../actions/answers';
+import { getLiveFeedbackChatsForQuestionId } from '../../selectors/liveFeedbackChats';
+
+interface ChipButtonIndicatorProps {
+  questionId: number;
+  syncStatus: keyof typeof GET_HELP_SYNC_STATUS;
+  setSyncStatus: Dispatch<SetStateAction<keyof typeof GET_HELP_SYNC_STATUS>>;
+}
+
+const translations = defineMessages({
+  syncingWithCodaveri: {
+    id: 'course.assessment.submission.GetHelpChatPage.syncingWithCodaveri',
+    defaultMessage: 'Preparing',
+  },
+  syncedWithCodaveri: {
+    id: 'course.assessment.submission.GetHelpChatPage.syncedWithCodaveri',
+    defaultMessage: 'Ready',
+  },
+  failedSyncingWithCodaveri: {
+    id: 'course.assessment.submission.GetHelpChatPage.failedSyncingWithCodaveri',
+    defaultMessage: 'Unavailable',
+  },
+});
+
+const GetHelpSyncIndicatorMap = {
+  Syncing: {
+    color: 'default' as const,
+    icon: <LoadingIndicator bare size={20} />,
+    label: translations.syncingWithCodaveri,
+  },
+  Synced: {
+    color: 'success' as const,
+    icon: <CheckCircle />,
+    label: translations.syncedWithCodaveri,
+  },
+  Failed: {
+    color: 'error' as const,
+    icon: <Cancel />,
+    label: translations.failedSyncingWithCodaveri,
+  },
+};
+
+const ChipButton = (props: ChipButtonIndicatorProps): JSX.Element | null => {
+  const { questionId, syncStatus, setSyncStatus } = props;
+
+  const submissionId = getSubmissionId();
+  const { t } = useTranslation();
+  const dispatch = useAppDispatch();
+
+  const liveFeedbackChats = useAppSelector((state) =>
+    getLiveFeedbackChatsForQuestionId(state, questionId),
+  );
+
+  const currentThreadId = liveFeedbackChats?.currentThreadId;
+
+  const createChat = (): void => {
+    dispatch(createLiveFeedbackChat({ submissionId, questionId }))
+      .then(() => setSyncStatus(GET_HELP_SYNC_STATUS.Synced))
+      .catch(() => setSyncStatus(GET_HELP_SYNC_STATUS.Failed));
+  };
+
+  const fetchChatStatus = (): void => {
+    dispatch(
+      fetchLiveFeedbackStatus({
+        submissionId,
+        questionId,
+        threadId: currentThreadId,
+      }),
+    )
+      .then(() => setSyncStatus(GET_HELP_SYNC_STATUS.Synced))
+      .catch(() => setSyncStatus(GET_HELP_SYNC_STATUS.Failed));
+  };
+
+  const createChatOnDemand = (): void => {
+    if (!currentThreadId) {
+      createChat();
+    } else {
+      fetchChatStatus();
+    }
+  };
+
+  useEffect(() => {
+    createChatOnDemand();
+  }, [currentThreadId]);
+
+  const chipProps = GetHelpSyncIndicatorMap[syncStatus];
+  if (!chipProps) return null;
+
+  return (
+    <Chip
+      clickable={syncStatus === GET_HELP_SYNC_STATUS.Failed}
+      color={chipProps.color}
+      icon={chipProps.icon}
+      label={t(chipProps.label)}
+      onClick={() => {
+        if (syncStatus === GET_HELP_SYNC_STATUS.Failed) {
+          setSyncStatus(GET_HELP_SYNC_STATUS.Syncing);
+          createChatOnDemand();
+        }
+      }}
+      size="medium"
```

Comment:
nit: size="small" and `self-center`

Reply:
done (I also resized the LoadingIndicator from 20 to 15 to accommodate this)